Content.Server/Projectiles/ProjectileSystem.cs

```diff
@@ -71,18 +71,21 @@ private void OnStartCollide(EntityUid uid, ProjectileComponent component, ref St
 
         component.DamagedEntity = true;
 
-        if (component.CanPenetrate)
+        if (component.DeleteOnCollide)
+            QueueDel(uid);
+
+        if (component.DamageAfterCollide)
         {
             component.DamagedEntity = false;
 
-            if (component.DeleteOnCollide && !HasComp<MobStateComponent>(target))
-                QueueDel(uid);
-        }
-        else if (component.DeleteOnCollide)
-        {
-            QueueDel(uid);
+            if (component.CanOnlyPenetrateMobs)
+            {
+                if (!HasComp<MobStateComponent>(target))
+                    QueueDel(uid);
+            }
         }
```

Comment:
This control flow is also messy and shouldn't be on projectilecomponent as it's bloating it too much.